corgi/core/models.py

```diff
@@ -190,8 +190,7 @@ def desc(self):
         return self.desc.name
 
     def save(self, *args, **kwargs):
-        if not self.purl:
-            self.purl = self.obj.purl
+        self.purl = self.obj.purl
```

Comment:
I found that sometimes self.obj is None which causes this to fail. Let's add a if self.obj check here.

Reply:
That would indicate that there are dangling nodes without actual objects that they should be pointing to. I think in general we lack some sort of deletion mechanism for nodes when their objects get deleted. I think we always assumed that we'd be appending data only and never deleting so it would be interesting to know how a ComponentNode came to be orphaned.

Reply:
You are right, never mind it. I could be caused by the data cleanup loop. It should never happen in normal operations.

Reply:
for completeness ... 

root cause is likely due to change near models.py#170  via

https://github.com/RedHatProductSecurity/component-registry/pull/129/commits/37f70af40b62b90c074ce91d2d0a7ac4249ae4df

the change itself is perfectly valid - though the existence of these nodes before the change would have needed a custom migration to clean up.

Reply:
Resolved by #145.